src/Stratis.Bitcoin.Features.Miner/Controllers/MiningController.cs

```diff
@@ -62,16 +39,17 @@ public MiningController(Network network, IPowMining powMining, ILoggerFactory lo
         /// <returns>List of block header hashes of newly mined blocks.</returns>
         /// <remarks>It is possible that less than the required number of blocks will be mined because the generating function only
         /// tries all possible header nonces values.</remarks>
-        [Route("generate")]
+        [Route("startmining")]
         [HttpPost]
-        public IActionResult Generate([FromBody]MiningRequest request)
+        public IActionResult StartMining([FromBody]MiningRequest request)
         {
             Guard.NotNull(request, nameof(request));
 
             try
             {
-                if (this.network.Consensus.IsProofOfStake)
-                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.MethodNotAllowed, "Method not allowed", "Method not available for Proof of Work");
+                if (this.fullNode.Network.Consensus.IsProofOfStake &&
```

Comment:
Interesting, but not required that much becuase such blocks will anyway be rejected.

Reply:
Yes but we WANT to be able to mine on a POS network (see PR description) before LastPowBLock height ;) So there this check is needed otherwise we just block mining if the network's IsProofOfStake setting is true. This is to help Lukasz's testing ;)